sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeFormatter.scala

```diff
@@ -35,11 +34,11 @@ private class CodeFormatter {
   private var indentString = ""
 
   private def addLine(line: String): Unit = {
-    val indentChange = line.count(_ == '{') - line.count(_ == '}')
+    val indentChange = line.count(c => c == '{' || c == '(') - line.count(c => c == '}' || c == ')')
```

Comment:
You could do `"({[".indexOf(c) >= 0` and support square brackets too if you want.